modules/codegen-plugin/src/sbt-test/codegen-plugin/aws-newtype-flatten/src/main/scala/Main.scala

```diff
@@ -0,0 +1,20 @@
+
+import com.amazonaws.dynamodb.ArchivalSummary
+import com.amazonaws.dynamodb.BackupArn
+import smithy4s.Timestamp
+
+object Main extends App {
+
+  val archivalSummary = new ArchivalSummary(
+    // archivalDateTime needs to be rewritten from a com.amazonaws.dynamodb.Date to a simple timestamp
+    archivalDateTime = Some(Timestamp.fromEpochSecond(java.time.Instant.now().getEpochSecond)),
+
+    // archivalReason needs to be rewritten from com.amazonaws.dynamodb.ArchivalReason to a String
+    archivalReason = Some("This is just a string"),
+
+    // archivalBackupArn remains as a newtype because com.amazonaws.dynamodb.BackupArn
+    // has the @length trait applied
```

Comment:
**thought:** perhaps we should flatten these too but add the constraints to the members? @baccata

Reply:
I think the newtype is justified when a constraint is present.